src/components/header/header.test.js

```diff
@@ -4,39 +4,26 @@ import Header from './header';
 import { DEFAULT_SUBREDDIT } from '../../constants';
 
 describe('header', () => {
-  let header = null;
-  let spyHistoryPush = null;
-
-  beforeEach(() => {
-    header = render(<Header />);
-    spyHistoryPush = jest.spyOn(header.history, 'push');
-  });
-
-  afterEach(() => {
-    spyHistoryPush.mockRestore();
-  });
-
   it('routes how-it-works link', () => {
-    const { getByRole } = header;
-
-    fireEvent.click(getByRole('link', { name: /how it works/i }));
+    const { getByRole } = render(<Header />);
 
-    expect(spyHistoryPush).toHaveBeenCalledWith('/#how-it-works');
+    expect(getByRole('link', { name: /how it works/i })).toHaveAttribute('href', '/#how-it-works');
   });
 
   it('routes about link', () => {
-    const { getByRole } = header;
-
-    fireEvent.click(getByRole('link', { name: /about/i }));
+    const { getByRole } = render(<Header />);
```

Comment:
You can import `screen` from `@testing-library/react` and use `screen.getByRole` instead. This is valid for all the queries. The advantage is that you have auto-completion in your editor and you don't need to update this line every time you need to use another query.